File: 백준/Silver/7568. 덩치/덩치.py
import sys
import copy

N = int(sys.stdin.readline())
graph = [[] * 2 for _ in range(N)]

rank = []
result = []

for i in range(N):
    graph[i] = list(map(int, sys.stdin.readline().split()))
    
for i in range(N):
    count = 0
    kg = graph[i][0]
    cm = graph[i][1]
    for j in range(N):
        if kg < graph[j][0] and cm < graph[j][1]:
            count += 1
    rank.append(count+1)
    
for i in range(N):
    print(rank[i], end=" ")

File: 백준/Silver/7568. 덩치/덩치.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(bf.readLine());
        StringBuilder sb = new StringBuilder();

        List<Human> list = new ArrayList<>();

        for(int i = 0; i<N; i++){
            String[] sa = bf.readLine().split(" ");
            int weight = Integer.parseInt(sa[0]);
            int height = Integer.parseInt(sa[1]);

            list.add(new Human(weight, height));
        }

        for(int i=0; i<list.size(); i++){
            Human human = list.get(i);
            int count = 0;
            for(int j=0; j<list.size(); j++){
                if(i == j) continue;
                if(human.height < list.get(j).height && human.weight < list.get(j).weight) count++;
            }
            human.winCount = count;
        }

        for(int i=0; i<list.size(); i++){
            sb.append(list.get(i).winCount + 1).append(" ");
        }

        System.out.println(sb);
    }

}

class Human{
    int weight;
    int height;
    int winCount;
    int ranking;

    public Human(int weight, int height){
        this.weight = weight;
        this.height = height;
    }
}
